test/parser/parser-errors.t

```diff
@@ -125,14 +125,57 @@ mtt: Parse error: Boxed expression is expected
   mtt: Parse error: Missing or unexpected lexeme in parenthesized expression
   [1]
 
-$ mtt parse -e "let x = () x"
-mtt: Parse error: Missing or unexpected lexeme in parenthesized expression
-[1]
+  $ mtt parse -e "let x = () x"
+  mtt: Parse error: Missing or unexpected lexeme in parenthesized expression
+  [1]
 
-$ mtt parse -e "let x = () y = x"
-mtt: Parse error: Missing or unexpected lexeme in parenthesized expression
-[1]
+  $ mtt parse -e "let x = () y = x"
+  mtt: Parse error: ZERO token can be used in match-expression only
+  [1]
 
-$ mtt parse -e "let x = (fun z: A. z) y = () in x y"
-mtt: Parse error: Missing or unexpected lexeme in parenthesized expression
-[1]
+  $ mtt parse -e "let x = (fun z: A. z) y = () in x y"
+  mtt: Parse error: ZERO token can be used in match-expression only
+  [1]
+
+### Match-expression
+  $ mtt parse <<EOF
+  > match () with
+  > end
+  > EOF
+  mtt: Parse error: Empty branch in match-expression
+  [1]
+
+  $ mtt parse <<EOF
+  > match x with
+  > | zero
+  > end
+  > EOF
+  mtt: Parse error: Empty branch in match-expression
+  [1]
+
+  $ mtt parse <<EOF
+  > match x
+  > | zero
+  > end
+  > EOF
+  mtt: Parse error: Missing "with" keyword
+  [1]
+
+  $ mtt parse <<EOF
+  > match x with
+  > | zero => ()
+  > EOF
+  mtt: Parse error: Empty branch in match-expression
+  [1]
+
+  $ mtt parse <<EOF
+  > match x with
+  > | zero => ()
+  > | succ n => ()
```

Comment:
Can we also have an error message for the case when the two branches of pattern-matching are swapped?
```
match x with
| succ n => ()
| zero => ()
end

Reply:
It seems menhir doesn't generate this case. It returns the same error message as in the case `expr_eof: MATCH UNIT WITH PIPE WITH`.